fawltydeps/packages.py

```diff
@@ -116,68 +117,80 @@ class UserDefinedMapping(BasePackageResolver):
+            else:
+                result[normalised_name] = value
+        return result
+
     @property
     @calculated_once
     def packages(self) -> Dict[str, Package]:
         """Gather a custom mapping given by a user.
 
         Mapping may come from two sources:
-        * _mapping_path_ which is a file in a given path, which is parsed to a ditionary
+        * _mapping_paths_ which is a set of files in a given path, which is parsed to a ditionary
         * _custom_mapping_ which is a dictionary of package to imports mapping.
 
         This enumerates the available packages  _once_, and caches the result for
         the remainder of this object's life in _packages.
         """
-        custom_mapping_from_file: Dict[str, List[str]] = {}
+        custom_mapping_from_files: Dict[str, List[str]] = {}
+
+        if self.mapping_paths is not None:
+            logger.debug(f"Loading user-defined mapping from {self.mapping_paths}")
+            for path in self.mapping_paths:
+                with open(path, "rb") as mapping_file:
+                    custom_mapping_from_files = self.add_mapping_dicts(
+                        custom_mapping_from_files, tomllib.load(mapping_file)
+                    )
 
-        if self.mapping_path is not None:
-            logger.debug(f"Loading user-defined mapping from {self.mapping_path}")
-            with open(self.mapping_path, "rb") as mapping_file:
-                custom_mapping_from_file = tomllib.load(mapping_file)
-
-        mapping = {
-            Package.normalize_name(name): Package(
-                name,
-                {DependenciesMapping.USER_DEFINED: set(imports)},
-            )
-            for name, imports in custom_mapping_from_file.items()
-        }
+        custom_mapping = custom_mapping_from_files
 
         if self.custom_mapping is not None:
             logger.debug("Applying user-defined mapping from settings.")
 
-            for name, imports in self.custom_mapping.items():
-                normalised_name = Package.normalize_name(name)
-                if normalised_name in mapping:
-                    logger.info(
-                        "Mapping for %s already found in %s. Import names "
-                        "from the configuration file are appended to ones "
-                        "found in the mapping file.",
-                        normalised_name,
-                        self.mapping_path,
-                    )
-                    mapping[normalised_name].add_import_names(
-                        *imports, mapping=DependenciesMapping.USER_DEFINED
-                    )
-                else:
-                    mapping[normalised_name] = Package(
-                        name,
-                        {DependenciesMapping.USER_DEFINED: set(imports)},
-                    )
+            custom_mapping = self.add_mapping_dicts(
+                self.custom_mapping, custom_mapping_from_files
+            )
```

Comment:
With my suggested `accumulate_mappings()` above, I think we can make this both lazier and more efficient by passing a lazily-computed iterator of mappings to a single `accumulate_mappings()` call:

```python
        def _custom_mappings() -> Iterator[CustomMapping]:
            if self.custom_mapping is not None:
                logger.debug("Applying user-defined mapping from settings.")
                    yield self.custom_mapping

            if self.mapping_paths is not None:
                for path in self.mapping_paths:
                    logger.debug(f"Loading user-defined mapping from {path}")
                    with open(path, "rb") as mapping_file:
                        yield tomllib.load(mapping_file)

        custom_mapping = accumulate_mappings(_custom_mappings())
```

Reply:
applied